### PR TITLE
fix(Core): "Silenced - You can only chat with GMs" spell.

### PR DESCRIPTION
**Pull request by masterking32**
Fix spell : [Silenced](https://wotlk.evowow.com/?spell=1852)

### How to test:
`.aura 1852` and try to chat as normal/other players and with GMs.

### CODE REVIEW

**Comment by BarbzYHOOL** [comment]
looks good, i suppose players usually only talk with GMs using whispering

File: src/server/game/Handlers/ChatHandler.cpp [hunk 1]
```diff
@@ -236,14 +243,6 @@ void WorldSession::HandleMessagechatOpcode(WorldPacket & recvData)
             sender->UpdateSpeakTime(specialMessageLimit);
     }
 
-    // pussywizard: optimization
```
Comment (BarbzYHOOL) [inline src/server/game/Handlers/ChatHandler.cpp]: how was this an optimization???

Comment (masterking32) [reply src/server/game/Handlers/ChatHandler.cpp]: That wasn't an optimization xD

File: src/server/game/Handlers/ChatHandler.cpp [hunk 2]
```diff
@@ -386,11 +385,11 @@ void WorldSession::HandleMessagechatOpcode(WorldPacket & recvData)
                 }
 
             // pussywizard: optimization
-            /*if (GetPlayer()->HasAura(1852) && !receiver->IsGameMaster())
```
Comment (FrancescoBorzi) [inline src/server/game/Handlers/ChatHandler.cpp]: I'd remove the above comment too

File: src/server/game/Handlers/ChatHandler.cpp (hunk 1 above)
Comment (BarbzYHOOL) [reply src/server/game/Handlers/ChatHandler.cpp]: it's just that nobody use this spell but you I guess xD

Comment (Helias) [reply src/server/game/Handlers/ChatHandler.cpp]: exactly, if you don't use that spell, it's an "optimization" removing that condition, but only if you are sure that you do not use never the related spell.

Comment (BarbzYHOOL) [reply src/server/game/Handlers/ChatHandler.cpp]: so we reintroduce it or not? xD

Comment (masterking32) [reply src/server/game/Handlers/ChatHandler.cpp]: I already use this spell, When I see a advertiser in my server xD

**Comment by gengarshadowball** [comment]
Good PR to mute players for an x amount of time. Make a mod that implements a timer on the players mute using this spell? :thinking:

**Comment by Si1ker** [comment]
Changes proposed included,

Test performed: Player with aura cannot speak, cannot chat by party/raid or whisp to any other player but GMs. Working as intended, nice work 👍 

2020-03-03 23_32_59-Window

**Comment by BarbzYHOOL** [comment]
gonna destroy our performances

(joking)

**Pull request merged**